app/src/main/java/com/github/se/eventradar/ui/home/Home.kt

```diff
@@ -183,7 +188,7 @@ fun HomeScreen(
                       start.linkTo(parent.start)
                       end.linkTo(parent.end)
                     })
-        (uiState.eventList.allEvents.isEmpty()) ->
+        (uiState.viewList && uiState.upcomingEventList.allEvents.isEmpty()) ->
```

Comment:
this is not a very good use of the when block. This logic should be split. One if/when statement for dealing with whether the list or the map is shown, and another for what will be shown based on the uiState. Having them together like this is not very readable.

Reply:
done

Reply:
After checking Sonar Cloud, this change actually makes our cognitive complexity higher. Reverting the changes.